packages/rum-core/src/common/config-service.js

```diff
@@ -240,15 +241,19 @@ class Config {
   }
 
   getLocalConfig() {
-    let config = sessionStorage.getItem(LOCAL_CONFIG_KEY)
+    let config = localStorage.getItem(LOCAL_CONFIG_KEY)
     if (config) {
       return JSON.parse(config)
     }
   }
 
-  setLocalConfig(config) {
+  setLocalConfig(config, merge) {
     if (config) {
-      sessionStorage.setItem(LOCAL_CONFIG_KEY, JSON.stringify(config))
+      if (merge) {
+        const prevConfig = this.getLocalConfig()
+        config = { ...prevConfig, ...config }
+      }
+      localStorage.setItem(LOCAL_CONFIG_KEY, JSON.stringify(config))
```

Comment:
Any reason for not using a cookie? They are easier to maintain and reason about. one issue i see here with local storage is, we never clean up the key from the user's machine which is bad and with cookie you set expires then its cleaned up by the browser.

Reply:
There are a few problems with using cookies here but the main issue is that, we don't want this data to be sent to the server on every request. Essentially we would be trading network bandwidth for a little bit of local storage, which I don't is a good idea.

Reply:
You mean sending session value on every APM request?  Thats a single header value on every request and headers are compressed with newer protocols like HTTP2 and QUIC which is too small to even reason about. The main issue here is we are leaving this data on the user's machine which is more concerning IMO. If we can find a way to delete this data, we are good with either one of them.

Reply:
Generally, I don't think it's a good idea to send data on the network, if that data only is needed locally.
In this case, we should investigate ways to delete the data, rather than sending these configs on each request.

Reply:
Discussed this in the meeting, we're going to use localStorage since we don't want to send unnecessary data over network on every request.